File: langgraph/data/semantic_equiv/4.py
from typing import List

def calculate_mean_abs_dev(values: List[float]) -> float:
    average = sum(values) / len(values)
    deviations = [abs(value - average) for value in values]
    return sum(deviations) / len(values)

# method name: mean_absolute_deviation -> calculate_mean_abs_dev
# parameter name: numbers -> values
# variable name: mean -> average, counts -> deviations
# replaced generator expression with list comprehension for clarity

File: langgraph/result/qwen/7b-eq2js/4.js
function calculateMeanAbsDev(values) {
    const average = values.reduce((acc, val) => acc + val, 0) / values.length;
    const deviationSum = values.reduce((acc, val) => acc + Math.abs(val - average), 0);
    return deviationSum / values.length;
}

// method name: mean_absolute_deviation -> calculateMeanAbsDev
// parameter name: numbers -> values
// variable name: mean -> average, counts -> deviationSum
// changed the order of operations slightly by storing the sum of deviations in a variable